log4j-api/src/main/java/org/apache/logging/log4j/internal/LogBuilder.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache license, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
+package org.apache.logging.log4j.internal;
+
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.message.Message;
+
+/**
+ * Class Description goes here.
+ */
+public interface LogBuilder {
+
+    public static final LogBuilder INSTANCE = new LogBuilder() {};
```

Comment:
I think this would be easier to follow if this was named something along the lines of `NOP`, `LogBuilder.INSTANCE` sounds a lot like the DefaultLogBuilder.

Should these be default methods? While it's nice to avoid causing breaks when we add new methods, it's unfortunate that we will be potentially throwing data away.

Reply:
Yes, NOOP_LOGBUILDER would probably be better.

Actually, the chance of throwing data away is minimal. AbstractLogger implements the methods that virtually all Loggers use.